### PR TITLE
Adds Zonal Diffusion

### PR DESCRIPTION
**Pull request by HenryDane**
This pull request adds zonal advection-diffusion, zonal heat diffusion, and zonal moist diffusion. Additionally, this makes the adv-diff solver support periodic boundary conditions. 

I've also added some unit tests to verify that the changes work.

Please let me know if you have any comments or feedback or questions.

### CODE REVIEW

**Comment by brian-rose** [comment]
Hi @HenryDane, this looks awesome! Thanks for the contribution, and especially for including tests with your PR! It may take me a bit of time to digest everything you've done here.

File: climlab/tests/test_advdiff_procs.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from __future__ import division
+import numpy as np
+import climlab
+import pytest
+
```
Comment (brian-rose) [inline climlab/tests/test_advdiff_procs.py:5]:
```suggestion
from climlab import constants as const
```
should fix the errors currently showing up in the test suite.

**Review by brian-rose — CHANGES_REQUESTED** [review]

**Comment by HenryDane** [comment]
I committed the change you suggested -- thanks very much for pointing that out to me. It looks like all the tests are passing now.

**Comment by brian-rose** [comment]
> I committed the change you suggested -- thanks very much for pointing that out to me. It looks like all the tests are passing now.

Great! I'll take a closer look at the code when I get a chance.

**Comment by HenryDane** [comment]
I've been adding commits here ([f256da](https://github.com/climlab/climlab/pull/179/commits/6f256da735ca38e600b94873eed11920c5781075) through [4a6ba5b](https://github.com/climlab/climlab/pull/179/commits/4a6ba5baae84c5208b83a23898e400a6c25a41f0)) to fix an issue I noticed with this new version where if a model had both zonal and meridional adv-diff processes, the fluxes (`diffusive_flux`, `total_flux`, `advective_flux`, `flux_convergence`) for one adv-diff process could get over-written by others. 

These new commits fix this problem by adding an optional `diag_suffix` so that when a process inherits from `AdvectionDiffusion`, it can declare a suffix for the aforementioned attributes, thereby fixing the issue. 

With these changes, I believe that zonal advection-diffusion, zonal heat diffusion, and zonal moist diffusion should now work correctly both on their own and in the presence of other adv-diff processes.

**Comment by brian-rose** [comment]
@HenryDane can you merge in the latest changes from main branch into this PR? That will clear up the unrelated test failures.

**Comment by HenryDane** [comment]
Yes, of course. I just applied the merge and all the tests seem to be passing now.
